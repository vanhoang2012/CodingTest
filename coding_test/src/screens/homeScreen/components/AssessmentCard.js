import moment from 'moment';
import React from 'react';
import {Image, StyleSheet, View, Text} from 'react-native';
import Images from '../../../assets/images/Images';
import Colors from '../../../themes/Colors';
import globalStyle from '../../../themes/globalStyle';
import {
  responsiveFont,
  responsiveHeight,
  responsiveWidth,
} from '../../../themes/Layouts';

function AssessmentCard({item}) {
  //////////////////////////////////////////////////
  //////////////////// Variable ////////////////////
  //////////////////////////////////////////////////

  ///////////////////////////////////////////////////
  //////////////////// useEffect ////////////////////
  ///////////////////////////////////////////////////

  //////////////////////////////////////////////////
  //////////////////// Function ////////////////////
  //////////////////////////////////////////////////

  //////////////////////////////////////////////////
  ////////////////////// View //////////////////////
  //////////////////////////////////////////////////
  return (
    <View style={styles.container}>
      <Image
        resizeMode={'contain'}
        source={{uri: item?.image}}
        style={styles.image}
      />
      <View style={styles.content}>
        <View>
          <Text style={styles.title}>{item?.title}</Text>
          <Text style={styles.deadline}>
            {item?.deadline &&
            moment(item?.deadline).isValid() &&
            moment(item?.deadline).diff(moment(), 'day') <= 7
              ? moment(item?.deadline).diff(moment(), 'day') + ' days left'
              : ''}
          </Text>
        </View>
        <View style={styles.rowBetween}>
          <View>
            <Text style={styles.pointText}>Earn up to</Text>
            <Text style={styles.point}>
              {item?.point} <Text style={styles.pointText}>pts</Text>
            </Text>
          </View>
          <Image source={Images.arrowRight} style={styles.icon} />
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: responsiveWidth(16),
    width: responsiveWidth(270),
    backgroundColor: Colors.white,
    ...globalStyle.shadow,
    marginVertical: responsiveHeight(16),
    marginHorizontal: responsiveWidth(4),
    borderRadius: responsiveWidth(8),
  },
  image: {
    height: responsiveWidth(90),
    width: responsiveWidth(50),
  },
  title: {
    fontSize: responsiveFont(14),
    color: Colors.blackText,
    fontWeight: 'bold',
  },
  deadline: {
    fontSize: responsiveFont(12),
    color: Colors.blackText3,
    fontWeight: 'normal',
    marginTop: responsiveHeight(6),
  },
  content: {
    paddingLeft: responsiveWidth(16),
    flex: 1,
    justifyContent: 'space-between',
  },
  rowBetween: {
    flexDirection: 'row',
    alignItems: 'flex-end',
    justifyContent: 'space-between',
  },
  pointText: {
    fontSize: responsiveFont(12),
    fontWeight: 'normal',
    color: Colors.blackText3,
  },
  point: {
    fontSize: responsiveFont(14),
    fontWeight: 'bold',
    color: Colors.primary,
  },
  icon: {
    height: responsiveWidth(20),
    width: responsiveWidth(20),
    tintColor: Colors.red2,
  },
});

export default AssessmentCard;
