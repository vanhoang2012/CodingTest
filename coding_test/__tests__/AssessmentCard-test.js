/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import AssessmentCard from '../src/screens/homeScreen/components/AssessmentCard';

it('renders correctly', () => {
  const tree = renderer.create(
    <AssessmentCard
      item={{
        id: '1',
        title: 'Advanced Health Screening',
        point: 9999,
        image:
          'https://w7.pngwing.com/pngs/64/886/png-transparent-human-hand-and-pink-heart-illustration-health-care-medicine-computer-icons-hospital-icon-love-purple-text-thumbnail.png',
        deadline: '2021/12/24',
      }}
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('renders correctly 2', () => {
  const tree = renderer.create(
    <AssessmentCard
      item={{
        id: '1',
        title: 'Advanced Health Screening',
        image:
          'https://w7.pngwing.com/pngs/64/886/png-transparent-human-hand-and-pink-heart-illustration-health-care-medicine-computer-icons-hospital-icon-love-purple-text-thumbnail.png',
        deadline: '2021/12/24',
      }}
    />,
  );
  expect(tree).toMatchSnapshot();
});
