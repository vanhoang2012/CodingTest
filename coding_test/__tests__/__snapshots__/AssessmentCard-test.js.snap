// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 16,
      "elevation": 2,
      "flexDirection": "row",
      "marginHorizontal": 8,
      "marginVertical": 26.285714285714285,
      "padding": 32,
      "shadowColor": undefined,
      "shadowOffset": Object {
        "height": 1.6428571428571428,
        "width": 1.6428571428571428,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
      "width": 540,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://w7.pngwing.com/pngs/64/886/png-transparent-human-hand-and-pink-heart-illustration-health-care-medicine-computer-icons-hospital-icon-love-purple-text-thumbnail.png",
      }
    }
    style={
      Object {
        "height": 180,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
        "paddingLeft": 32,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#2a2a2a",
            "fontSize": 28,
            "fontWeight": "bold",
          }
        }
      >
        Advanced Health Screening
      </Text>
      <Text
        style={
          Object {
            "color": "#7f7f7f",
            "fontSize": 24,
            "fontWeight": "normal",
            "marginTop": 9.857142857142858,
          }
        }
      >
        3 days left
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#7f7f7f",
              "fontSize": 24,
              "fontWeight": "normal",
            }
          }
        >
          Earn up to
        </Text>
        <Text
          style={
            Object {
              "color": "#04577F",
              "fontSize": 28,
              "fontWeight": "bold",
            }
          }
        >
          9999
           
          <Text
            style={
              Object {
                "color": "#7f7f7f",
                "fontSize": 24,
                "fontWeight": "normal",
              }
            }
          >
            pts
          </Text>
        </Text>
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/right-arrows.png",
          }
        }
        style={
          Object {
            "height": 40,
            "tintColor": "#B71743",
            "width": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly 2 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 16,
      "elevation": 2,
      "flexDirection": "row",
      "marginHorizontal": 8,
      "marginVertical": 26.285714285714285,
      "padding": 32,
      "shadowColor": undefined,
      "shadowOffset": Object {
        "height": 1.6428571428571428,
        "width": 1.6428571428571428,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
      "width": 540,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://w7.pngwing.com/pngs/64/886/png-transparent-human-hand-and-pink-heart-illustration-health-care-medicine-computer-icons-hospital-icon-love-purple-text-thumbnail.png",
      }
    }
    style={
      Object {
        "height": 180,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
        "paddingLeft": 32,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#2a2a2a",
            "fontSize": 28,
            "fontWeight": "bold",
          }
        }
      >
        Advanced Health Screening
      </Text>
      <Text
        style={
          Object {
            "color": "#7f7f7f",
            "fontSize": 24,
            "fontWeight": "normal",
            "marginTop": 9.857142857142858,
          }
        }
      >
        3 days left
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#7f7f7f",
              "fontSize": 24,
              "fontWeight": "normal",
            }
          }
        >
          Earn up to
        </Text>
        <Text
          style={
            Object {
              "color": "#04577F",
              "fontSize": 28,
              "fontWeight": "bold",
            }
          }
        >
           
          <Text
            style={
              Object {
                "color": "#7f7f7f",
                "fontSize": 24,
                "fontWeight": "normal",
              }
            }
          >
            pts
          </Text>
        </Text>
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/right-arrows.png",
          }
        }
        style={
          Object {
            "height": 40,
            "tintColor": "#B71743",
            "width": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;
